pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/painless/UpsertPainless.java

```diff
@@ -46,35 +50,52 @@ public void export() {
 
         Throwable ioException = null;
         for (int tryNum = 1; tryNum < 3; tryNum++) {
-            ioException = executeExport(clientInstance, updateByQueryRequest, tryNum);
+            ioException = executeExport(clientInstance, updateByQueryRequest);
             if (ioException == null) {
                 break;
             }
+            logger.info("Error occurred exporting data to ES on try number {} (retrying): {}", tryNum, ioException.toString());
         }
         if (ioException != null) {
             logger.error("Unable to connect to ElasticSearch: {}", ioException.getMessage());
             throw new DsmInternalError("Unable to connect to ElasticSearch", ioException);
         }
     }
 
-    private Throwable executeExport(RestHighLevelClient clientInstance, UpdateByQueryRequest updateByQueryRequest, int i) {
+    private Throwable executeExport(RestHighLevelClient clientInstance, UpdateByQueryRequest updateByQueryRequest) {
```

Comment:
Any advantage in returning Exception. 
This method can throw the Exception and the caller can catch it in try..catch , right ?

Reply:
Yes, I like your suggestion better than what I have here.
This approach is historical and I was grounded in it rather than seeing a better way.
Thx!

Reply:
Actually, I see now why I did it that way. The loop with a catch is doable, but kind of clumsy and hard to read. The most straightforward implementation of that type is a `for` loop with integer counters, but the code inside the loop has to check for the loop max count to decide whether to throw or try again. Alternatively, the exception could be assigned to a variable outside of the loop, but at that point returning the throwable is easier to understand.

What do you think? Are you ok with this implementation or did you have something specific in mind?